packages/compass/src/app/utils/telemetry.ts

```diff
@@ -115,15 +115,15 @@ async function getHostnameForConnection(
         return null;
       }
     );
-    if (!uri) {
-      return null;
+    if (uri) {
+      connectionStringData = new ConnectionString(uri, {
+        looseValidation: true,
+      });
     }
-    connectionStringData = new ConnectionString(uri, {
-      looseValidation: true,
-    });
   }
 
-  return connectionStringData.hosts[0];
+  const [hostname] = (connectionStringData.hosts[0] ?? '').split(':');
```

Comment:
I feel like this is code I've written elsewhere as well, but keep in mind that the hostname itself can also contain `:` if it is an IPv6 address

Reply:
Can it be Atlas host as well with an IPv6 address?

Reply:
And another question, did we solve this in this other project or do we keep doing the same? I am wondering if we want to address it here or this is fine leave as is.

Reply:
Note: As discussed in Slack we want to handle IPv6.